src/appengine/handlers/testcase_detail/show.py

```diff
@@ -361,15 +361,47 @@ def _get_blob_size_string(blob_key):
   return utils.get_size_string(blob_size)
 
 
+def _format_reproduction_help(reproduction_help):
+  """Format a reproduction help string as HTML."""
+
+  def _linkify(plain_text_input):
+    """Convert links in a plain text string to HTML."""
+    if (not plain_text_input.startswith('http://') and
+        not plain_text_input.startswith('https://')):
+      return plain_text_input
+
+    return '<a href="{url}">{url}</a>'.format(url=plain_text_input)
+
+  if not reproduction_help:
+    return ''
+
+  result = ''
+  for line in reproduction_help.splitlines():
+    # Add a break before each line except the first.
+    if result:
+      result += '<br>'
+
+    input_parts = [_linkify(part) for part in line.split(' ')]
+    result += ' '.join(input_parts)
+
+  return result
```

Comment:
I think django and jinja2 provides the urlize and linebreaksbr filters?

Reply:
Since these are getting supplied to our polymer template we don't have easy access to filters. This doesn't feel great here, though. It might be better to implement a similar function in javascript.

Reply:
Can we call it from python from here: https://github.com/pallets/jinja/blob/master/jinja2/utils.py#L190 ?

Reply:
Good call. Done!